src/containers/ChangePasswordPage.jsx

```diff
@@ -0,0 +1,221 @@
+  'defaultprofile:addresses:element',
+  'defaultprofile:addresses:billingaddresses:default',
+  'defaultprofile:paymentmethods',
+  'defaultprofile:paymentmethods:paymenttokenform',
+  'defaultprofile:paymentmethods:element',
+  'passwordresetform',
+];
+
+class ChangePasswordForm extends React.Component {
+  static propTypes = {
+    location: ReactRouterPropTypes.location.isRequired,
+    history: ReactRouterPropTypes.history.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      oldPassword: '',
+      newPassword: '',
+      newPasswordConfirmed: '',
+      failedSubmit: false,
+      email: '',
+      passwordResetUri: '',
+    };
+    this.setOldPassword = this.setOldPassword.bind(this);
+    this.setNewPassword = this.setNewPassword.bind(this);
+    this.setNewPasswordConfirmed = this.setNewPasswordConfirmed.bind(this);
+    this.submitNewPassword = this.submitNewPassword.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchProfileData();
+  }
+
+  setOldPassword(event) {
+    this.setState({ oldPassword: event.target.value });
+  }
+
+  setNewPassword(event) {
+    this.setState({ newPassword: event.target.value });
+  }
+
+  setNewPasswordConfirmed(event) {
+    this.setState({ newPasswordConfirmed: event.target.value });
+  }
+
+  cancel() {
+    const { location, history } = this.props;
+    if (location.state && location.state.returnPage) {
+      history.push(location.state.returnPage);
+    } else if (localStorage.getItem(`${Config.cortexApi.scope}_oAuthRole`) === 'REGISTERED') {
+      history.push('/profile');
+    } else {
+      history.push('/');
+    }
+  }
+
+  submitNewPassword(event) {
+    event.preventDefault();
+    const {
+      oldPassword, newPassword, newPasswordConfirmed, email, passwordResetUri,
+    } = this.state;
+    if (!oldPassword || !newPassword || !newPasswordConfirmed) {
+      this.setState({ failedSubmit: true });
+    }
+    if (passwordResetUri) {
+      login()
+        .then((response) => {
+          cortexFetch(`${passwordResetUri}/?zoom=resetpasswordaction`, {
+            method: 'post',
+            headers: {
+              'Content-Type': 'application/json',
+              Authorization: localStorage.getItem(`${Config.cortexApi.scope}_oAuthToken`),
+            },
+            body: JSON.stringify({ 'user-id': email }),
+          })
+            .then(() => {});
+          return response;
+        })
+        .catch((error) => {
+          // eslint-disable-next-line no-console
+          console.error(error.message);
+        });
+    }
+  }
+
+  fetchProfileData() {
+    login().then(() => {
+      cortexFetch(`/?zoom=${zoomArray.join()}`,
+        {
+          headers: {
+            'Content-Type': 'application/json',
+            Authorization: localStorage.getItem(`${Config.cortexApi.scope}_oAuthToken`),
+          },
+        })
+        .then(res => res.json())
+        .then((res) => {
+          if (res && res._passwordresetform) {
```

Comment:
This piece should be how we can validate if the forgot password feature is even available. We'll have to add a call to the appmodallogin component to make a call like this on componentdidmount and display the "forgot password" button if `passwordresetform` is available.